blockchain/blockchain.go

```diff
@@ -50,13 +55,33 @@ func init() {
 			ethClient = ethclient.NewClient(rawClient)
 		}
 
+		agentAddress := common.HexToAddress(config.GetString(config.AgentContractAddressKey))
+
 		// Setup agent
-		if a, err := NewAgent(common.HexToAddress(config.GetString(config.AgentContractAddressKey)), ethClient); err != nil {
+		if a, err := NewAgent(agentAddress, ethClient); err != nil {
 			log.WithError(err).Panic("error instantiating agent")
 		} else {
 			agent = a
 		}
 
+		// Determine "version" of agent contract and set local signature hash creator
+		if bytecode, err := ethClient.CodeAt(context.Background(), agentAddress, nil); err != nil {
+			log.WithError(err).Panic("error retrieving agent bytecode")
+		} else {
+			bcSum := md5.Sum(bytecode)
+			if bytes.Equal(bcSum[:], []byte{244, 176, 168, 6, 74, 56, 171, 175, 38, 48, 245, 246, 189, 0, 67, 200}) {
+				sigHasher = func(i []byte) []byte {
+					return crypto.Keccak256([]byte{0x19}, []byte("Ethereum Signed Message:\n32"),
```

Comment:
Suggest making these `[]byte("Ethereum ...)` things top-level vars/const (they can't be const as-is because they're slices) to clarify that these don't change based on input